app/commands/convene.tsx

```diff
@@ -0,0 +1,158 @@
+import { format } from "date-fns";
+import { Message } from "discord.js";
+import type { MessageContextMenuInteraction, TextChannel } from "discord.js";
+import { ContextMenuCommandBuilder } from "@discordjs/builders";
+import { ApplicationCommandType } from "discord-api-types/v10";
+
+import { reacord } from "~/discord/client";
+import { quoteAndEscape } from "~/helpers/discord";
+import { ReportReasons, reportUser } from "~/helpers/modLog";
+import { resolutions } from "~/helpers/modResponse";
+
+import { fetchSettings, SETTINGS } from "~/models/guilds.server";
+import { applyRestriction, ban, kick, timeout } from "~/models/discord.server";
+import { ModResponse } from "~/commands/reacord/ModResponse";
+
+export const command = new ContextMenuCommandBuilder()
+  .setName("Convene mods")
+  .setType(ApplicationCommandType.Message);
+
+export const handler = async (interaction: MessageContextMenuInteraction) => {
+  const { targetMessage: message, member, guild } = interaction;
+  if (!(message instanceof Message) || !member || !guild) {
+    return;
+  }
+
+  const { modLog, moderator } = await fetchSettings(guild, [
+    SETTINGS.modLog,
+    SETTINGS.moderator,
+  ]);
+
+  const logChannel = (await guild.channels.fetch(modLog)) as TextChannel;
+  if (!logChannel || !logChannel.isText()) {
+    throw new Error("Failed to load mod channel");
+  }
+
+  const { message: logMessage } = await reportUser({
+    message,
+    reason: ReportReasons.mod,
+    extra: `‼️ <@${interaction.user.id}> requested mods respond`,
+  });
+
+  if (logMessage.hasThread) {
+    return;
+  }
+
+  const thread = await logMessage.startThread({
+    name: `${message.author.username} mod response ${format(new Date(), "P")}`,
+  });
+  const originalChannel = (await message.channel.fetch()) as TextChannel;
+  const instance = await reacord.send(
+    thread.id,
+    <ModResponse
+      modRoleId={moderator}
+      onResolve={async (resolution) => {
+        instance.deactivate();
+        switch (resolution) {
+          case resolutions.restrict:
+            reportUser({
+              reason: ReportReasons.mod,
+              message,
+              extra: "✅ Restricted",
+            });
+            await applyRestriction(message.member!);
+            message.reply(
```

Comment:
This is only useful if they're misbehaving in a restricted channel, yeah?

Reply:
In the immediate term sure, but I'm thinking that there will be a lively conversation in these threads before the outcome is determined. Figure this can be a replacement for us talking in private chats about if someone has earned a restriction